### PR TITLE
 refactor `removeBucketReplication` api to ts

### PR DESCRIPTION
**Pull request by prakashsvmx**
"refactor `removeBucketReplication` api to ts" 

Try to remove the bucket replication 

```js
try {
  await s3Client.removeBucketReplication('source-bucket')
  console.log('Success')
} catch (e) {
  return console.log(e)
}
```

<details>

<summary> Steps to setting up a bucket replication </summary>

```
# Setup Bucket replication e.g b/w 2 sites
Site 1 (local22)
---------
CI=true  MINIO_ROOT_USER=minio MINIO_ROOT_PASSWORD=minio123 minio server /tmp/sem{1...4}  --address ":22000" --console-address ":9025"

Site 2(local23)
---------------------------
CI=true  MINIO_ROOT_USER=minio MINIO_ROOT_PASSWORD=minio123 minio server /tmp/semc{1...4}  --address ":23000" --console-address ":9035"

mc mb local22/source-bucket
mc mb local23/dest-bucket
mc version enable local22/source-bucket
mc version enable local23/dest-bucket

mc replicate add local22/source-bucket --remote-bucket http://minio:minio123@localhost:23000/dest-bucket --priority 1 

echo "Numeric" | mc pipe local22/source-bucket
mc ls -r local22/source-bucket
mc ls -r local23/dest-bucket

mc replicate ls local22/source-bucket

```
</details>

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -581,26 +581,26 @@ minioClient.getBucketReplication('bucketname', function (err, replicationConfig)
...
 **Example**
+Please refer to : [remove-bucket-replication.mjs](..%2Fexamples%2Fremove-bucket-replication.mjs)
 
 ```js
-minioClient.removeBucketReplication('bucketname', function (err, replicationConfig) {
-  if (err) {
-    return console.log(err)
-  }
+try {
+  minioClient.removeBucketReplication('source-bucket')
```
Comment (trim21) [inline docs/API.md:599]: await

File: src/internal/client.ts [hunk 2]
```diff
@@ -887,4 +887,14 @@ export class TypedClient {
     const xmlResult = await readAsString(httpRes)
     return xmlParsers.parseListBucket(xmlResult)
   }
+
+  removeBucketReplication(bucketName: string, callback: NoResultCallback): void
```
Comment (trim21) [inline src/internal/client.ts:891]: missing async overload. The impl is not emit in type declaration

Comment (prakashsvmx) [reply src/internal/client.ts:892]: Thank you so much for the valuable feedback :+1:

File: docs/API.md (hunk 1 above)
Comment (trim21) [inline docs/API.md:599]: Not changed

**Review by kaankabalak — APPROVED** [review]
LGTM

**Comment by trim21** [comment]
lgtm

**Pull request merged**